ownCloud/Settings/MediaUploadSettingsSection.swift

```diff
@@ -72,6 +155,180 @@ class MediaUploadSettingsSection: SettingsSection {
 
 		self.add(row: convertPhotosSwitchRow!)
 		self.add(row: convertVideosSwitchRow!)
+
+		// Instant upload requires at least one configured account
+		OCBookmarkManager.shared.loadBookmarks()
+		if OCBookmarkManager.shared.bookmarks.count > 0 {
+			instantUploadPhotosRow = StaticTableViewRow(switchWithAction: { [weak self] (_, sender) in
+				if let convertSwitch = sender as? UISwitch {
+					self?.changeAndRequestPhotoLibraryAccessForOption(optionSwitch: convertSwitch, completion: { (switchState) in
+						self?.userDefaults.instantUploadPhotos = switchState
+						self?.updateDynamicUI()
+
+						self?.userDefaults.instantUploadPhotosAfter = switchState ? Date() : nil
+
+						NotificationCenter.default.post(name: UserDefaults.MediaUploadSettingsChangedNotification, object: nil)
+					})
+				}
+				}, title: "Instant Upload Photos".localized, value: self.userDefaults.instantUploadPhotos)
+
+			instantUploadVideosRow = StaticTableViewRow(switchWithAction: { [weak self] (_, sender) in
+				if let convertSwitch = sender as? UISwitch {
+					self?.changeAndRequestPhotoLibraryAccessForOption(optionSwitch: convertSwitch, completion: { (switchState) in
+						self?.userDefaults.instantUploadVideos = switchState
+						self?.updateDynamicUI()
+
+						self?.userDefaults.instantUploaVideosAfter = switchState ? Date() : nil
+
+						NotificationCenter.default.post(name: UserDefaults.MediaUploadSettingsChangedNotification, object: nil)
+					})
+				}
+				}, title: "Instant Upload Videos".localized, value: self.userDefaults.instantUploadVideos)
+
+			bookmarkSelectionRow = StaticTableViewRow(valueRowWithAction: { [weak self] (_, _) in
+				self?.showAccountSelectionViewController()
+				}, title: "Account".localized, value: "", accessoryType: .disclosureIndicator, identifier: MediaUploadSettingsSection.bookmarkSelectionRowIdentifier)
+
+			uploadPathSelectionRow = StaticTableViewRow(valueRowWithAction: { [weak self] (_, _) in
+				self?.showUploadPathSelectionViewController()
+				}, title: "Upload Path".localized, value: "", accessoryType: .disclosureIndicator, identifier: MediaUploadSettingsSection.pathSelectionRowIdentifier)
+
+			self.add(row: instantUploadPhotosRow!)
+			self.add(row: instantUploadVideosRow!)
+		}
+
+		updateDynamicUI()
+	}
+
+	private func getSelectedBookmark() -> OCBookmark? {
```

Comment:
Not needed for now, but would have been cool to have a kind of standardized/generic picker for an account + path combination that could also be used in other places.